### PR TITLE
Release 0.14.0

### LINKED ISSUE #1369: New year, new release
Time to make a new release!

### LINKED ISSUE #1372: Make version 0.14 release.
## Description

As title says.

### PR DESCRIPTION
**Pull request by ni4**
Fixes #1372 
Fixes #1369

### CODE REVIEW

**Comment by ni4** [comment]
@dewyatt As from discussion at https://github.com/rnpgp/rnp/issues/1377 looks like I lacked knowledge on git's tags.
As I now understand, correct way would be to push the v0.14.0 tag only once this PR is merged, however before that CI doesn't run correctly since it was picking up the 0.13.1 tag, which doesn't work well with the `ruby-rnp`.

**Comment by dewyatt** [comment]
> @dewyatt As from discussion at #1377 looks like I lacked knowledge on git's tags.
> As I now understand, correct way would be to push the v0.14.0 tag only once this PR is merged, however before that CI doesn't run correctly since it was picking up the 0.13.1 tag, which doesn't work well with the `ruby-rnp`.

Yeah the release process could probably use some rethinking. Maybe there's some new GHA features we can look at to improve things.

**Comment by ni4** [comment]
> Yeah the release process could probably use some rethinking. Maybe there's some new GHA features we can look at to improve things.

What about to not look at the latest version tag, but at contents of version.txt instead (or, if it is absent - use commit timestamp)?

**Comment by ni4** [comment]
> What about to not look at the latest version tag, but at contents of version.txt instead (or, if it is absent - use commit timestamp)?

Ping @dewyatt , what do you think about this idea? Then we'll just push tag once release is made, and will not depend on it in CI/build files.

**Comment by ni4** [comment]
@ronaldtse @dewyatt Could you please check whether english in the changelog is good enough :) ? Thanks!

File: CHANGELOG.md
Comment (rrrooommmaaa) [inline CHANGELOG.md:3]: 01-04-2021 ?

File: CHANGELOG.md
Comment (antonsviridenko) [inline CHANGELOG.md:13]: I think it is commonly accepted that changelog should contain changes that matter to users, i.e. new features, bugs fixed, changes in public API. Internal code changes don't belong here, developers can look them in Git history if they are interested.

File: CHANGELOG.md
Comment (antonsviridenko) [inline CHANGELOG.md:49]: same here

File: CHANGELOG.md
Comment (antonsviridenko) [reply CHANGELOG.md:3]: why not  01-15-2021?

Comment (ni4) [reply CHANGELOG.md:3]: Thanks, updated.

File: CHANGELOG.md
Comment (ni4) [reply CHANGELOG.md:49]: Thanks, removed some too detailed lines. However I still think that we should leave lines about the Python 3/C++  to let users know that internals now are more modern.

**Review by rrrooommmaaa — APPROVED** [review]
LGTM!

File: CHANGELOG.md
Comment (ronaldtse) [reply CHANGELOG.md]: Can we change these dates to yyyy-mm-dd? 😉

Comment (ni4) [reply CHANGELOG.md]: @ronaldtse Sure, I'll update the PR.

**Review by antonsviridenko — APPROVED** [review]

**Comment by ni4** [comment]
@ronaldtse Is it good to go now? Or let's wait for @dewyatt review as well?

**Comment by ronaldtse** [comment]
Let's wait for @dewyatt to confirm. Thanks!

**Comment by dewyatt** [comment]
Hey guys I'm sort of locked out of (most) of my accounts (email for one) right now as I'm in the process of trying to redo security properly so I can start signing commits. I figured it was a good time since I have a bit of new hardware and new yubikeys and all. But it requires shuffling so much around that it's taking way longer than expected.

> > What about to not look at the latest version tag, but at contents of version.txt instead (or, if it is absent - use commit timestamp)?
> 
> Ping @dewyatt , what do you think about this idea? Then we'll just push tag once release is made, and will not depend on it in CI/build files.

So IIRC this is basically just swapping the current priority, right? Like right now we check for .git first, then version.txt. I would say let's try it. I know I put a lot of thought into the way it currently is but I can't recall a good reason for this particular aspect so we should probably give it a shot. I'll take a look at the versioning script.

**Comment by ni4** [comment]
> So IIRC this is basically just swapping the current priority, right? Like right now we check for .git first, then version.txt. I would say let's try it. I know I put a lot of thought into the way it currently is but I can't recall a good reason for this particular aspect so we should probably give it a shot. I'll take a look at the versioning script.

Yeah, since release tag should be pushed after the release PR is merged, but release PR needs to have latest version to run the CI.

P.S. As I understand you'll need some more time to review this PR/let it merge?

**Comment by dewyatt** [comment]
> > So IIRC this is basically just swapping the current priority, right? Like right now we check for .git first, then version.txt. I would say let's try it. I know I put a lot of thought into the way it currently is but I can't recall a good reason for this particular aspect so we should probably give it a shot. I'll take a look at the versioning script.
> 
> Yeah, since release tag should be pushed after the release PR is merged, but release PR needs to have latest version to run the CI.
> 
> P.S. As I understand you'll need some more time to review this PR/let it merge?

Yes let me check it out a bit more please.

**Review by dewyatt — APPROVED** [review]
Not sure if you want to add the remove-key commits or save that for next release but LGTM

**Comment by ni4** [comment]
@dewyatt I think it would be good enough if remove-key is added to the next release. Hopefully this time we'll not spend year to release the next version )

@ronaldtse Is it good to go now?

**Comment by ni4** [comment]
ping @ronaldtse

**Comment by ronaldtse** [comment]
All good. Thank you!

**Review by ronaldtse — APPROVED** [review]

**Comment by ni4** [comment]
Let's have v0.14.0 released!

**Pull request merged**
